pkg/termination/handler.go

```diff
@@ -87,7 +89,34 @@ func (h *handler) run(ctx context.Context, wg *sync.WaitGroup) error {
 	}
 	klog.V(1).Infof("Monitoring node %q for machine \"%s/%s\"", h.nodeName, machine.Namespace, machine.Name)
 
-	return fmt.Errorf("not implemented")
+	if err := wait.PollImmediateUntil(h.pollInterval, func() (bool, error) {
+		resp, err := http.Get(h.pollURL.String())
+		if err != nil {
+			return false, fmt.Errorf("could not get URL %q: %v", h.pollURL.String(), err)
+		}
+		switch resp.StatusCode {
+		case http.StatusNotFound:
+			// Instance not terminated yet
+			klog.V(2).Infof("Instance not marked for termination")
```

Comment:
should we prepend the machine/node here?

Reply:
Semi tempted to switch this over to the `logr` style structured logging that CAPI uses, WDYT?

Reply:
💯